lib/Camp/Master.pm

```diff
@@ -319,6 +320,234 @@ sub copy_paths_config_path {
+configuration tokens (see CAMP CONFIGURATION VARIABLES).  If not specified,
+parsing does not occur.
+
+=item I<script.allow_errors>
+
+If provided with a Perly true value, errors handling or executing the code in
+I<script.path> will be treated as warnings. Otherwise, handling or exeuction
+errors are fatal and abort the mkcamp/refresh-camp process.
+
+=back
+
+Because the YAML stream is converted to an array, the scripts are processed in
+order of specification. Thus, you can have dependencies between code execution
+order and they'll work out as long as you arrange them in the proper order in
+the file. However, if you do have such dependencies, it's important that they
+match up in context sections; otherwise, it is possible to have one script
+execute and another not if they do not share the same array of context
+sections.
+
+Here's an example configuration file:
+
+ ---
+ script:
+     path: bin/relative/to/camp/bar.pl
+     parse: 0
+     allow_errors: 1
+ context:
+     - db
+ ---
+ script:
+     path: /full/path/to/bin/foo.pl
+     parse: 0
+     allow_errors: 1
+ context:
+     - db
+     - vcs
+ ---
+ script:
+     path: bin/__CAMP_FOO__/with_token/always_run/baz.pl
+     parse: 1
+     allow_errors: 0
+ context:
+     - global
+ ---
+ script:
+     path: bin/mkcamp_and_refresh-camp/only/buz.pl
+     parse: 0
+     allow_errors: 0
+ context:
+     - mkcamp
+     - refresh-camp
+ ---
+ script:
+     path: /used/to/run/from/run_post_mkcamp_command.pl
+     parse: 0
+     allow_errors: 1
+ context:
+     - mkcamp
+
+=back
+
+=cut
+
+sub run_post_processing {
+    my %sections = map { $_ => 1, } grep { defined and /\S/ } (@_, 'global');
+
+    my $conf = config_hash();
+    my $file = post_processing_config_path();
+    if (! -f $file) {
+        print "no post-processing configuration file to run.\n";
+        return;
+    }
+
+    use_yaml();
+    my @data = parse_yaml( $file );
+    if (! @data) {
+        print "no commands found in post-processing file.\n";
+        return;
+    }
+
+    die "the post-processing data structures are invalid\n"
+        if grep {
+            not
+                ref ($_) eq 'HASH'
+                &&
+                length (($_->{script} || {})->{path} // '')
+                &&
+                ref ($_->{context}) eq 'ARRAY'
+        } @data;
+
+    SCRIPT:
+    for my $node (@data) {
+        my ($script, $context) = @{$node}{qw/script context/};
+        next SCRIPT unless
+            grep { $sections{$_} } @$context;
+
+        my $cmd = $script->{path};
+        $cmd = substitute_hash_tokens($cmd, $conf)
+            if $script->{parse};
+
+        $cmd = File::Spec->catfile( $conf->{path}, $cmd )
+            unless File::Spec->file_name_is_absolute($cmd);
+
+        {
+            local $@;
+            eval {
+                -e $cmd
+                    or die "File '$cmd' cannot be found\n";
+                -x $cmd
+                    or die "'$cmd' is not executable\n";
```

Comment:
I think those need `.\n` at the end so the message doesn't come out like:

```
File '/path/to/file' cannot be found Skipping.
```

Reply:
Done. Also noticed I had `chop` instead of `chomp` so changed that too.

Reply:
@msjohns1 Looks good!

@machack666 or anyone else watching this repo, do you have any comments before we merge this?

Reply:
Looks good here; so the only effective arg you get in the script is `$ENV{CAMP}`? (I guess you `use Camp::Config` in your script and get params that way.)

Reply:
@machack666 That's the way I understand it, yes.